Description: Fix the socket keepalive code.
Origin: upstream; http://git.libwebsockets.org/cgi-bin/cgit/libwebsockets/commit?id=bb66ac6ec13092d5f7ef2f8a199605fc95e25cba
Author: Andy Green <andy.green@linaro.org>
Last-Update: 2013-02-23

commit bb66ac6ec13092d5f7ef2f8a199605fc95e25cba
Author: Larry Hayes <lhayesg@gmail.com>
Date:   Fri Feb 22 09:16:20 2013 +0800

    keepalive swap interval and probes
    
    The interval and probes members for setting keepalive options
    per-connection on Linux were swapped.
    
    Signed-off-by: Larry Hayes <lhayesg@gmail.com>

--- a/lib/libwebsockets.c
+++ b/lib/libwebsockets.c
@@ -575,12 +575,12 @@
 					     (const void *)&optval, optlen) < 0)
 			return 1;
 
-		optval = context->ka_probes;
+		optval = context->ka_interval;
 		if (setsockopt(fd, IPPROTO_IP, TCP_KEEPINTVL,
 					     (const void *)&optval, optlen) < 0)
 			return 1;
 
-		optval = context->ka_interval;
+		optval = context->ka_probes;
 		if (setsockopt(fd, IPPROTO_IP, TCP_KEEPCNT,
 					     (const void *)&optval, optlen) < 0)
 			return 1;
